src/Bielu.Examine.Umbraco/Indexers/ElasticSearchUmbracoIndex.cs

```diff
@@ -46,7 +45,7 @@ public class BieluExamineUmbracoIndex(string? name, ILoggerFactory loggerFactory
         /// </summary>
 
         public bool PublishedValuesOnly => CurrentContentValueSetValidator?.PublishedValuesOnly ?? false;
-        private IContentValueSetValidator? CurrentContentValueSetValidator => indexOptions.CurrentValue.Validator as IContentValueSetValidator;
+        private IContentValueSetValidator? CurrentContentValueSetValidator => _namedOptions.Validator as IContentValueSetValidator;
```

Comment:
we used current value to ensure it always load current value, as .net App might not restarted by just changing values.

Reply:
Copied from https://github.com/umbraco/Umbraco-CMS/blob/6e3a6917c8264b51caf7d3bf56907caeec0f223e/src/Umbraco.Examine.Lucene/UmbracoContentIndex.cs#L34

The indexOptions.CurrentValue does not contain the right values. Please check this for ExternalIndex by setting a breakpoint.

Reply:
not sure why umbraco is doing things soo overcomplicated 🙄